### DISCUSSION TITLE
[Question] How to delete records from a data table using data from a column, python flet

### DISCUSSION THREAD

**Discussion by DiegoMarin1** [flet-dev/flet#1992] [Q&A]
### Question

I want to know how I can delete a record from my table using data from a column, not by number of columns.

this is my table

And I fill it with data from a database

Example I would like to delete a record if its data in the id column is 550

### Code sample

```python
self.tablaCilindros = DataTable(
        columns=[
            DataColumn(Text("id")),
            DataColumn(Text("Empresa")),
            DataColumn(Text("Tamano")),
            DataColumn(Text("Pico")),
            DataColumn(Text("Accion"))
        ],
        rows=[

        ]
    )

def quitar(self, pagee):
    pagee = pagee

    #What this does is eliminate records according to the position 
    #of the row number, and I want it to delete by id that I get in the database

    del self.tablaCilindros.rows[ExampleNumber1]
    pagee.update()
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ingdesarr1** [comment]
if the ID in the datatable column is the same ID in the database, you can get all your rows, consult the first cell, that correspond to the ID, and ask for the required value that you need. Once you have found the value, and the row respectively, you can use pop or some other method, and then update the table.

**Reply by DiegoMarin1** [reply]
The bad thing about this case is that this is not the case, because that table shows the data of many people and each one has different IDs, that is, the number of rows in the table never match the database IDs, for example. I want that to be eliminated by a single piece of data. of the identification column. For example, that person has 5 cylinders assigned, and their ids that are in the database are (500, 501, 502, 503, 504) then there is the problem that the id does not match the index of the rows and it appears that is out of range

**Reply by DiegoMarin1** [reply]
![imagen_2023-10-26_125428669](https://github.com/flet-dev/flet/assets/112134946/901a639d-7fdf-4698-bb82-62a32f84d24e)

That is the program, I have 5 people registered, each of them has several cylinders

![imagen_2023-10-26_130200486](https://github.com/flet-dev/flet/assets/112134946/69ac4c38-cf8e-411f-bfde-535d25651e48)

So as you will see there are two datatables, the orange one is the one I am asking in this question, the program does the following if you select a record from the orange table it changes to the green one, what I cannot do is that when the information is passed to the green list is removed from the orange list

I already tried to delete by index and it works, but when I exit and return the orange table it remains the same with all the records, I already have an incomplete function that will do the following: as soon as you select a person and redirect you to that screen, the table will be filled with the data associated with that person in that table, if you have records in the green table that have the same id as the orange table, I want to delete the elements in the orange table that have the same id as the green table, and there is no how to do it

I don't know if you understand what I want to explain to you, but I need help with this and I've tried and I can't think of anything.

**Reply by ingdesarr1** [reply]
A ver si entiendo mejor. Las dos tablas, naranja y verde. Si presiono en una fila de la tabla naranja, esta cambia a la tabla verde, pero persiste en la tabla naranja. Eso quiere decir que ya conoces la posición de la fila dentro de la tabla y consultas la información y demás. Se me ocurre que, cuando selecciones la fila y hagas el cambio, en ese momento, cambies la propiedad visible de la fila a False. De esa forma, la fila en general no sera visible en la tabla verde. Si la información esta siendo traida desde una base de datos, y actualiza la tabla desde la base de datos, siempre se van a mantener los registros, a menos que mueva los datos dentro de la base de datos, o coloque un nuevo campo que permite hacerlo visible o no, y consultarlo para añadir las filas a la tabla. 

Si no es asi, quedo atento.

**Reply by lucenajj** [reply]
Good morning Diego, how are you? I've been trying to put my cards like yours for days, could you help me please?

**Comment by ndonkoHenri** [comment]
Please build a simple working example and I will help.
(I made something similar [here](https://github.com/flet-dev/flet/discussions/949#discussioncomment-4809623) , but for the listview)
